src/Core/Utilities/CoreHelpers.cs

```diff
@@ -264,5 +265,37 @@ public static T Clone<T>(T obj)
         {
             return JsonConvert.DeserializeObject<T>(JsonConvert.SerializeObject(obj));
         }
+
+        public static string TextColorFromBgColor(string hexColor, int threshold = 166)
+        {
+            var bgColorObj = new ColorConverter().ConvertFromString(hexColor);
+            if (bgColorObj == null)
+            {
+                return "#ff000000";
+            }
+            var bgColor = (Color)bgColorObj;
+            var luminance = bgColor.R * 0.299 + bgColor.G * 0.587 + bgColor.B * 0.114;
+            return luminance > threshold ? "#ff000000" : "#ffffffff";
+        }
+
+        public static string StringToColor(string str)
+        {
+            if (str == null)
+            {
+                return "#33ffffff";
+            }
```

Comment:
I think that given that now this method is in `CoreHelpers` (and could be reused) we shouldn't hardcode the default value inside but return null directly and let the caller handle the default value:
```c#
public static string StringToColor(string str, string default)
{
    if (str == null)
    {
        return null;
    }
```
so when calling it like in `AvatarImageSource`, we should do so like this:
```c#
var bgColor = CoreHelpers.StringToColor(_id ?? upperCaseText) ?? "#33ffffff";
```

Reply:
Agreed, good call